pandas/core/frame.py

```diff
@@ -6446,7 +6449,7 @@ def sort_index(
 
     @overload
     def sort_index(
-        self,
+        self: DataFrameT,
```

Comment:
as far as I remember, pyright doesn't like self annotations that are broader than the class itself (cannot use NDFrameT).